src/engine/sync/enginesync.cpp

```diff
@@ -60,11 +60,18 @@ void EngineSync::requestSyncMode(Syncable* pSyncable, SyncMode mode) {
     }
     case SyncMode::Follower: {
         // A request for follower mode may be converted into an enabling of soft
-        // leader mode.
-        activateFollower(pSyncable);
+        // leader mode if this still be best choice
+        if (m_pLeaderSyncable == pSyncable) {
+            // Avoid that pickLeader() returns pSyncable with SyncMode::LeaderExplicit
```

Comment:
this code feels like a workaround for a bug.  Am I right that just calling activateFollower(pSyncable) is not working because that syncable is already explicit leader?  I think this code should remain the same and the edge case should be fixed elsewhere